modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/Pipe.java

```diff
@@ -277,22 +288,33 @@ public ByteBuffer copyAndProduce(final ContentDecoder decoder) throws IOExceptio
         try {
             ByteBuffer duplicate = null;
             setInputMode(buffer);
+            int totalRead = 0;
```

Comment:
shall we rename this variable to totalBytesRead?

Reply:
Fixed in 1a225662c71267f47edbbee23cb3dcf7c2106fd3